### PR TITLE
Get docs build passing

### PR DESCRIPTION
**Pull request by jrbourbeau**

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -3,11 +3,12 @@ version: 2
...
   os: ubuntu-22.04
   tools:
-    python: "3.11"
+    python: "3.12"
 
 sphinx:
   configuration: docs/source/conf.py
-  fail_on_warning: true
+  # Temporarily turning off to get docs build passing
+  # fail_on_warning: true
```
Comment (jrbourbeau) [inline .readthedocs.yaml:11]: The docs build was failing due to this warning (seemingly from Python itself) being elevated to an error 

```
/Users/james/mambaforge/envs/dask-docs-py312/lib/python3.12/functools.py:docstring of functools.CreationDispatch.register_inplace.<locals>.decorator.<locals>.wrapper:10: WARNING: Inline emphasis start-string without end-string.
```

I haven't been able to track down where this is coming from yet and resolve things properly, so I'm just temporarily removing out "elevate docs warnings to errors" settings to at least get the docs build passing somehow.

**Review by jrbourbeau — COMMENTED** [review]
I'm going to merge this in order to get our docs build passing, but will open an issue to track turning `fail_on_warning` back on

**Review by phofl — COMMENTED** [review]
Lgtm for now

**Comment by jrbourbeau** [comment]
Here's the related issue https://github.com/dask/dask/issues/11193

**Pull request merged**